### DISCUSSION TITLE
PDF Thumbnail

### DISCUSSION THREAD

**Discussion by RolRodr** [CollectionBuilder/collectionbuilder.github.io#96] [General]
I wonder if it would be a good idea to turn the first page of a PDF into its thumbnail rather than having a PDF-icon placeholder? That would allow PDFs to also be featured in features like the carousal or the timeline. 

Just a thought! Also apologies if this is not the correct avenue for positing suggestions. Feel free to take it down if so.

**Comment by evanwill** [comment]
Thanks for the question @RolRodr! Since the static template doesn't have a server or back end to do any processing of the media, CB-GH is constrained to what files you can add in manually. For CB-GH, the idea is minimal template with simple set up of the media items--so there is no processing for thumbnails or other derivatives images. This just keeps it straightforward to put together a collection without needing to do any extra steps. 

This does have limitations, like the idea you are mentioning here, and that the images displayed aren't optimized for their context. So the alternative is our more robust template, [CollectionBuilder-CSV](https://github.com/CollectionBuilder/collectionbuilder-csv)! CB-CSV has the option for thumb and small image representations for all items in the collection (no matter the media type). There is even an automation Rake task to generate thumbs for PDFs and images that you can use in the template. But all of that is more to learn (and install)! So generally we see CB-GH as a starting point and learning tool, and if you are interesting in going further, you can move into CB-CSV (everything is compatible, you will just need to add a few more fields in the metadata). 

Checkout our [walkthroughs](https://collectionbuilder.github.io/cb-docs/docs/walkthroughs/) on CB-CSV and derivatives if you want to see what it looks like.

**Comment by evanwill** [comment]
@RolRodr I am going to transfer this Issue to our Discussion forum so that other people can find it if they have similar questions!
